virtio-devices/src/vhost_user/blk.rs

```diff
@@ -483,4 +483,23 @@ impl Migratable for Blk {
             Ok(MemoryRangeTable::default())
         }
     }
+
+    fn migration_complete(&mut self) -> std::result::Result<(), MigratableError> {
+        // Make sure the device thread is killed in order to prevent from
+        // reconnections to the socket.
+        if let Some(kill_evt) = self.common.kill_evt.take() {
+            kill_evt.write(1).map_err(|e| {
```

Comment:
As we decide to shutdown the whole source VM after live-migration, we should not need to kill the vhost-user threads manually, right?

Reply:
Yes we do because for each vhost-user device we have a thread responsible for reconnecting if the connection drops. And since the shutdown path resumes every device before to shut them down, we can't afford to let vhost-user threads to reconnect just right before they get killed.